xee/ext.py

```diff
@@ -679,31 +679,19 @@ def _slice_collection(self, image_slice: slice) -> ee.Image:
     # Get the right range of Images in the collection, either a single image or
     # a range of images...
     start, stop, stride = image_slice.indices(self.shape[0])
-
-    # If the input images have IDs, just slice them. Otherwise, we need to do
-    # an expensive `toList()` operation.
-    if self.store.image_ids:
-      imgs = self.store.image_ids[start:stop:stride]
+    selectors = list(range(start, stop, stride))
+    col = self.store.image_collection.select(self.variable_name)
+    if self.shape[0] <= 5000:  # 5000 == max bands in an Image
+      col_as_image = col.toBands()
+      return col_as_image.select(selectors)
+    elif stop < 5000:  # 5000 == max bands in an Image
```

Comment:
Wont the stride effect this number?

Reply:
I don't think so. Even if the start/stride arguments would make the final image have fewer than 5000 bands, I'm avoiding `toList` by converting to a multiband image before slicing so I need to grab the first "stop" bands and `toBands` will fail if it would return an image with more than 5000 bands.